### PR TITLE
set attr num_sampled <= range_max

### PR DESCRIPTION
**Pull request by shkr**
For Log Uniform Candidate sampler the num_sampled cannot be more than the range_max as mentioned in #4112

### CODE REVIEW

**Comment by shkr** [comment]
I signed it!

**Comment by danmane** [comment]
@tensorflow-jenkins test this please

**Comment by zffchen78** [comment]
I think a better fix is to add a validation check on 
third_party/tensorflow/core/kernels/candidate_sampler_ops.cc:113

OP_REQUIRES(context, num_samples_ < range_max, ...)

**Comment by shkr** [comment]
@zffchen78 I am unable to find third_party/tensorflow. Can you provide a link to the file where you think the validation will be better off ?

**Comment by danmane** [comment]
@shkr look here: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/candidate_sampler_ops.cc

**Comment by shkr** [comment]
Thanks @danmane I will commit the change

**Comment by shkr** [comment]
@danmane let me know if any other changes are required

**Comment by martinwicke** [comment]
ping @danmane

File: tensorflow/core/kernels/candidate_sampler_ops.cc [hunk 1]
```diff
@@ -134,6 +134,7 @@ class AllCandidateSamplerOp : public BaseCandidateSamplerOp {
       : BaseCandidateSamplerOp(context) {
     int64 range_max;
     OP_REQUIRES_OK(context, context->GetAttr("num_sampled", &range_max));
+    CHECK_LE(num_sampled_, range_max);
```
Comment (zffchen78) [inline tensorflow/core/kernels/candidate_sampler_ops.cc]: Can you use OP_REQUIRES(ctx, num_sampled_, range_max, errors::InvalidArgument("...", )) instead? 

It'll also better to add a test case in the corresponding python test to check the expected exception is raised.

**Review by zffchen — CHANGES_REQUESTED** [review]

File: tensorflow/core/kernels/candidate_sampler_ops.cc (hunk 1 above)
Comment (shkr) [inline tensorflow/core/kernels/candidate_sampler_ops.cc]: Okay

**Comment by shkr** [comment]
@zffchen78 I have made the requisite change

**Comment by drpngx** [comment]
Jenkins, test this please.

**Comment by drpngx** [comment]
@zffchen78 does this look good now?

**Comment by zffchen78** [comment]
No. I requested a test.

**Comment by shkr** [comment]
@zffchen78 Did I miss anything or are you waiting for a jenkins test ?

**Comment by danmane** [comment]
@shkr I believe @zffchen78 means, he requests that you add a Python unit test (probably in python/kernel_tests) to verify the new behavior.

**Comment by shkr** [comment]
@danmane thanks I will add a unit test

**Comment by vrv** [comment]
Any updates?

**Comment by vrv** [comment]
Closing to help clear out our backlog, feel free to send a new PR once you've addressed these changes, or ping this thread and we'll reopen.

**Pull request closed without merging**